Assignment1_q1_permutations.py

```diff
@@ -0,0 +1,45 @@
+# Returns true if strings 's1' and 's2' are permutations of the other, false otherwise
+def is_permutation(s1, s2):
+
+    # We are case-insensitive
+    s1 = s1.lower()
+    s2 = s2.lower()
+
+    # If strings lengths are not equal they cannot be permutations of the other
+    if len(s1) != len(s2):
+        return False
+
+    # Initialise two dictionaries 'lc1' and 'lc2' to count occurrences of letters in 's1' and 's2'
+    lc1 = {}    # Letter count for 's1' (character count)
```

Comment:
Consider more descriptive names.
For example `character_counts1` or `char_dict1`.